src/ai/backend/manager/registry.py

```diff
@@ -4323,6 +4329,23 @@ async def _update_log() -> None:
         log_buffer.close()
 
 
+async def handle_vfolder_deletion(
+    context: AgentRegistry,
+    source: AgentId,
+    event: VFolderDeletionSuccessEvent | VFolderDeletionFailureEvent,
+) -> None:
+    match event:
+        case VFolderDeletionSuccessEvent(vfid=vfid):
+            await update_vfolder_status(
+                context.db, [vfid.folder_id], VFolderOperationStatus.DELETE_COMPLETE, do_log=True
+            )
+        case VFolderDeletionFailureEvent(vfid=vfid, message=message):
+            log.exception(f"Failed to delete vfolder (vfid:{vfid}, msg:{message})")
+            await update_vfolder_status(
+                context.db, [vfid.folder_id], VFolderOperationStatus.DELETE_ERROR, do_log=True
+            )
```

Comment:
Isn't it better to separate functions?

Reply:
We should decide whether to separate event handler functions or not

Reply:
We decided to separate such event handler functions into per-event functions